### DISCUSSION TITLE
Request Endpoint after received 401 Unauthorized response for token expired, using refresh_token cookie

### DISCUSSION THREAD

**Discussion by dhernandezgt** [usebruno/bruno#2973] [Q&A]
there is backend with response with 401 Unauthorized to a request which include a expired token.  Variables in Bruno APIClient are `token` and `refres_token` (saved as cookie).
The endpoint respose with: 
<img width="353" alt="image" src="https://github.com/user-attachments/assets/66a62cac-9fa2-45ea-a7ff-998e5f85d81f">
it is desired:

- after received `401 Unauthorized` send POST request to refresh URL including cookie in headers.
- refresh URL will response with new token and probably new refresh cooking (only if it is nearly to expired)
- save new token and refresh cookie, I guess it is bru.setVar(var,value)
- send new  GET request to required URL. 

```
console.log("Axios-send-request: Unauthorized response - Start");

const axios = require("axios");
let base = bru.getEnvVar('base');
const data = res.body;
const authCookie = bru.getVar('cookie');
let path = data.path;

if (res.status === 401) {
const urls = {
  required: base+path,
  refresh: base+'/dmz/refresh_token'
};
  console.log(urls)
  console.log(authCookie)
}
console.log("Axios-send-request: Unauthorized response - End");
```
please your help with a example or script that can use as a guide.

**Comment by iamsecuritec** [comment]
Hi Dhermandezgt,

Are you also facing an issue with unauthorized 401 errors?
I'm experiencing a similar problem. I have an application registered in Azure with the following API permissions: User.Read, openid, offline_access, and Mail.Read.

When I sign in through the web client, the app receives an access token valid for 60 minutes. After the 60 minutes pass, the application tries to send a refresh token request, but it gets a 401 Unauthorized response, and the user is automatically logged out.
